templates/_star_chart_v1.html

```diff
@@ -1,67 +1,113 @@
-                                data-modal-launch-observations="{{ observations|observation_pks }}"
-                            {% endif %}>
+                            {% for sublevel, observations in sublevels.items %}
+                                <td data-csl-id="{{ csl_id }}" style="border-bottom: 3pt solid black;"
+                                    class="text-center thicker"
+                                    {% if observations|length > 0 %}
+                                        data-modal-launch-observations="{{ observations|observation_pks }}"
+                                    {% endif %}>
 
-                            {% if observations|length > 0 %}
-                                <i data-toggle="tooltip" title="" class="fa fa-star"></i>
-                            {% endif %}
-                            {% if observations|length > 1 %}
-                                <span class="star-number">({{ observations|length }})</span>
-                            {% else %}
-                                &nbsp;
-                            {% endif %}
-                        </td>
-                    {% endfor %}
-                </tr>
+                                    {% if observations|length > 0 %}
+                                        <i data-toggle="tooltip" title="" class="fa fa-star"></i>
+                                    {% endif %}
+                                    {% if observations|length > 1 %}
+                                        <span class="star-number">({{ observations|length }})</span>
+                                    {% else %}
+                                        &nbsp;
+                                    {% endif %}
+                                </td>
+                            {% endfor %}
+                        </tr>
+                    {% else %}
+                        <tr class="construct-{{ construct.id }} class-{{ class.id }}">
+                            <th scope="col">&nbsp;</th>
+                            <th scope="row" class="student-{{ construct.id }} class-{{ class.id }}"
+                                style="white-space: nowrap; text-overflow: ellipsis; overflow: hidden; max-width: 150px;"
+                                >{{ student }}{% if not student.consented_to_research and not perms.kidviz.can_view_nonconsent_observations %}*{% endif %}
+                            </th>
+                    
+                            {% for sublevel, observations in sublevels.items %}
+                                <td data-csl-id="{{ csl_id }}"
+                                    class="text-center"
+                                    {% if observations|length > 0 %}
+                                        data-modal-launch-observations="{{ observations|observation_pks }}"
+                                    {% endif %}>
+
+                                    {% if observations|length > 0 %}
```

Comment:
I think that empty collection of observation returns false, so you don't have to calculate the length.